sql-plugin/src/main/scala/com/nvidia/spark/rapids/basicPhysicalOperators.scala

```diff
@@ -297,6 +297,22 @@ case class GpuRangeExec(range: org.apache.spark.sql.catalyst.plans.logical.Range
     throw new IllegalStateException(s"Row-based execution should not occur for $this")
 }
 
+object unionMetrics {
+  def apply(
+    batch: ColumnarBatch,
+    outputBatches: SQLMetric,
+    outputRows: SQLMetric,
+    totalTime: SQLMetric): ColumnarBatch = {
+    val nvtxRange = new NvtxWithMetrics("ProjectExec", NvtxColor.CYAN, totalTime)
```

Comment:
nit: Can we use `withResource` for this? it makes the code a little cleaner.

Reply:
Also this is a union operation, not a ProjectExec operation.

Reply:
Things missed during copy/paste  from other code :(